modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/index/BasicIndexTest.java

```diff
@@ -2393,4 +2459,68 @@ private enum SqlDataType {
             this.javaType = javaType;
         }
     }
+
+    /** */
+    static class TestH2Tree extends H2Tree {
```

Comment:
This approach looks good, but how can we be sure that scanCntr correlates with /* scanCount: 2 */ from issue ?